neuro_config_client/client.py

```diff
@@ -276,6 +283,299 @@ async def remove_storage(
+            ]
+            return np_templates
+
+    async def add_node_pool(
+        self,
+        cluster_name: str,
+        node_pool: NodePool,
+        *,
+        token: str | None = None,
+        start_deployment: bool = True,
+    ) -> Cluster:
+        """Add new node pool to the existing cluster.
+        Cloud provider should be already set up.
+
+        Make sure you use one of the available node pool templates,
+            if the cluster is deployed in public cloud (AWS / GCP / Azure / VCD).
+
+        Args:
+            cluster_name (str): Name of the cluster within the platform.
+            node_pool (NodePool): Node pool instance.
+                For templates, you could use template.to_node_pool() method
+            token (str | None, optional): User token to perform changes.
+                Should has write access to cluster://{cluster_name}/cloud_provider role.
+            start_deployment (bool, optional): Start applying changes. Defaults to True.
+
+        Returns:
+            Cluster: Cluster instance with applied changes
+        """
+        assert self._client
+
+        cluster = await self.get_cluster(cluster_name, token=token)
+        assert cluster.cloud_provider
+        if cluster.cloud_provider.type == CloudProviderType.ON_PREM:
+            return await self._add_onprem_node_pool(
+                cluster,
+                node_pool,
+                token=token,
+                start_deployment=start_deployment,
+            )
+        else:
+            return await self._add_cloud_node_pool(
+                cluster,
+                node_pool,
+                token=token,
+                start_deployment=start_deployment,
+            )
+
+    async def _add_onprem_node_pool(
+        self,
+        cluster: Cluster,
+        node_pool: NodePool,
+        token: str | None = None,
+        start_deployment: bool = True,
+    ) -> Cluster:
+        assert self._client
+        assert cluster.cloud_provider
+
+        url = self._clusters_url / cluster.name / "cloud_provider/node_pools"
+        headers = self._create_headers(token=token)
+        payload = self._payload_factory.create_node_pool(
+            node_pool, cluster.cloud_provider.type
+        )
+        async with self._client.post(
+            url.with_query(start_deployment=str(start_deployment).lower()),
+            headers=headers,
+            json=payload,
+        ) as response:
+            response.raise_for_status()
+            resp_payload = await response.json()
+            return self._entity_factory.create_cluster(resp_payload)
+
+    async def _add_cloud_node_pool(
+        self,
+        cluster: Cluster,
+        node_pool: NodePool,
+        *,
+        token: str | None = None,
+        start_deployment: bool = True,
+    ) -> Cluster:
+        assert self._client
+        assert cluster.cloud_provider
+        assert node_pool.id
+
+        np_templates = await self.get_node_pool_templates(
```

Comment:
This request is redundant, it's server responsibility to check that id/templates exist. Developer should pass correct id to the client. If developer needs to do these checks on the client side let him do it himself.
When I see function `add_node_pool` I don't expect it to do multiple requests on server, make validation on client. I expect it to make only one request to server with the data I provided and let the server decide if the data is correct.

Reply:
That makes sense, thanks for the clarification! Removed those steps in [d229e1f](https://github.com/neuro-inc/neuro-config-client/pull/83/commits/d229e1fabcc31e90a63c8084af36f6024a7e6780)